File: Python/chandu/Increasing_Order_Search_Tree.py
def solve(root):
    # CODE HERE
    kat = ans(root, [])
    res = t1 = TreeNode(-1)
    for i in range(len(kat)):
        if i< len(kat)-1:
            t1.val = kat[i]
            t1.right = TreeNode(-1)
            t1 = t1.right
        else:
            t1.val = kat[i]
    return res

def ans(root, kat):
    if root:
        ans(root.left,kat)
        kat.append(root.val)
        ans(root.right,kat)
    return kat

File: java/mohit/Trees-BST/Increasing Order Search Tree.java
// Dummy Node
static TreeNode curr = new TreeNode(-1);
static TreeNode ans = curr;

static TreeNode solve(TreeNode root)
{
    inorder(root);
    return ans.right;
}

static void inorder(TreeNode root)
{
    if(root == null) return;
    
    inorder(root.left);
    
    curr.right = root;
    root.left = null;
    curr = curr.right;
    
    inorder(root.right);
}